example/Agda/List.agda

```diff
@@ -0,0 +1,208 @@
+  foldr :
+    ∀ {Shape} {Pos} {a} {b} {i} →
+    Free Shape Pos
+    (Free Shape Pos a →
+     Free Shape Pos (Free Shape Pos b → Free Shape Pos b)) →
+    Free Shape Pos b →
+    Free Shape Pos (List Shape Pos a {i}) → Free Shape Pos b
+  foldr f e xs
+    = xs >>=
+      λ xs₁ →
+        case xs₁ of
+        λ { nil → e
+          ; (cons x xs') → f >>= λ f₁ → f₁ x >>= λ f₂ → f₂ (foldr f e xs')
+          }
+  foldr1 :
+    ∀ {Shape} {Pos} {a} →
+    ⦃ Partial Shape Pos ⦄ →
+    Free Shape Pos
+    (Free Shape Pos a →
+     Free Shape Pos (Free Shape Pos a → Free Shape Pos a)) →
+    Free Shape Pos (List Shape Pos a) → Free Shape Pos a
+  foldr1 f xs
+    = xs >>=
+      λ xs₁ →
+        case xs₁ of
+        λ { nil → error "foldr1: empty list"
+          ; (cons x xs') → foldr f x xs'
+          }
+  maximum :
+    ∀ {Shape} {Pos} →
+    ⦃ Partial Shape Pos ⦄ →
+    Free Shape Pos (List Shape Pos (ℤ Shape Pos)) →
+    Free Shape Pos (ℤ Shape Pos)
+  maximum x
+    = foldr1
+      (pure λ a → pure λ b → (a ≥ b) >>= λ f → if f then a else b) x
+  minimum :
+    ∀ {Shape} {Pos} →
+    ⦃ Partial Shape Pos ⦄ →
+    Free Shape Pos (List Shape Pos (ℤ Shape Pos)) →
+    Free Shape Pos (ℤ Shape Pos)
+  minimum x
+    = foldr1
+      (pure λ a → pure λ b → (a ≤ b) >>= λ f → if f then a else b) x
+  or :
+    ∀ {Shape} {Pos} →
+    Free Shape Pos (List Shape Pos (𝔹 Shape Pos)) →
+    Free Shape Pos (𝔹 Shape Pos)
+  or x = foldr (pure λ x₁ → pure λ x₂ → (x₁ ∨ x₂)) False x
+  product :
+    ∀ {Shape} {Pos} →
+    Free Shape Pos (List Shape Pos (ℤ Shape Pos)) →
+    Free Shape Pos (ℤ Shape Pos)
+  product x = foldr (pure λ x₁ → pure λ x₂ → (x₁ * x₂)) (pure 1) x
+  sum :
+    ∀ {Shape} {Pos} →
+    Free Shape Pos (List Shape Pos (ℤ Shape Pos)) →
+    Free Shape Pos (ℤ Shape Pos)
+  sum x = foldr (pure λ x₁ → pure λ x₂ → (x₁ + x₂)) (pure 0) x
+  foldl :
+    ∀ {Shape} {Pos} {b} {a} {i} →
+    Free Shape Pos
+    (Free Shape Pos b →
+     Free Shape Pos (Free Shape Pos a → Free Shape Pos b)) →
+    Free Shape Pos b →
+    Free Shape Pos (List Shape Pos a {i}) → Free Shape Pos b
+  foldl f e xs
+    = xs >>=
+      λ xs₁ →
+        case xs₁ of
+        λ { nil → e
+          ; (cons x xs') → foldl f (f >>= λ f₁ → f₁ e >>= λ f₂ → f₂ x) xs'
+          }
+  append :
+    ∀ {Shape} {Pos} {a} {i} →
+    Free Shape Pos (List Shape Pos a {i}) →
+    Free Shape Pos (List Shape Pos a) →
+    Free Shape Pos (List Shape Pos a)
+  append xs ys
+    = xs >>=
+      λ xs₁ →
+        case xs₁ of λ { nil → ys ; (cons x xs') → Cons x (append xs' ys) }
+  concat :
+    ∀ {Shape} {Pos} {a} →
+    Free Shape Pos (List Shape Pos (List Shape Pos a)) →
+    Free Shape Pos (List Shape Pos a)
+  concat x = foldr (pure λ x₁ → pure λ x₂ → append x₁ x₂) Nil x
+  and :
+    ∀ {Shape} {Pos} →
+    Free Shape Pos (List Shape Pos (𝔹 Shape Pos)) →
+    Free Shape Pos (𝔹 Shape Pos)
+  and x = foldr (pure λ x₁ → pure λ x₂ → (x₁ ∧ x₂)) True x
```

Comment:
Generated code should not be committed. Use the `.gitignore`d `./example/generated` folder for code generated from examples. We may also want to configure the CI pipeline to convert the examples to Agda automatically. This is blocked by the Agda tasks of #98 and #97. We should add a separate issue for the change of the CI pipeline and `./tool/full-test.sh` script.